### ISSUE TITLE
Strange video mode switching in Wing Commander III

### ISSUE THREAD

**Issue by ectorhga** [dosbox-staging/dosbox-staging#3659]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.82.0-alpha (b430c)

### What Operating System are you using?

Linux ARM (Raspberry Pi)

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

This is different from but similar to issue https://github.com/dosbox-staging/dosbox-staging/issues/3416. 

I run Dosbox Staging from the CLI without any window system in the background.

Whenever a scene changes (like when a video plays or I change rooms) I get a small 2-3 sec. blackout on my monitor. The reason for this are strange video mode changes. In the log file you can see that each time at first a "half video mode" (640x240;  as described in the issue above) is set and only then the full resolution (640x480) is set. My monitor seems to try setting a new video mode which leads to a small "backout".

I can run Dosbox Staging with `startx` and in a fixed window resolution (like 1920x1080), and the "blackouts" don't occur. Yet, the "half-video-mode" is still logged.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Start WC3 
2. Move around the ship

### Download URL of affected game or software

https://www.gog.com/de/game/wing_commander_3_heart_of_the_tiger

### Your configuration

```ini
[render]
aspect             = true
integer_scaling    = off
viewport           = fit
glshader           = none
```
```


```
```text
2024-05-08 11:32:04.508 | DISPLAY: VESA 640x480 256-colour graphics mode 101h at 70.007 Hz throttled VFR, scaled to 1440x1080 pixels with 1:1 (1:1) pixel aspect ratio
2024-05-08 11:32:04.541 | DISPLAY: VESA 640x240 256-colour graphics mode at 70.007 Hz throttled VFR, scaled to 1440x1080 pixels with 1:2 (1:2) pixel aspect ratio
2024-05-08 11:32:11.375 | DISPLAY: VESA 640x480 256-colour graphics mode 101h at 70.007 Hz throttled VFR, scaled to 1440x1080 pixels with 1:1 (1:1) pixel aspect ratio
2024-05-08 11:32:14.858 | DISPLAY: VESA 640x240 256-colour graphics mode at 70.007 Hz throttled VFR, scaled to 1440x1080 pixels with 1:2 (1:2) pixel aspect ratio
2024-05-08 11:32:21.191 | DISPLAY: VESA 640x480 256-colour graphics mode 101h at 70.007 Hz throttled VFR, scaled to 1440x1080 pixels with 1:1 (1:1) pixel aspect ratio
```
```


### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by weirddan455** [comment]
I believe the mode switching is normal behavior for this game.  It runs at 640X240 while it's playing a video and 640X480 otherwise.

>  I run Dosbox Staging from the CLI without any window system in the background.

This is a cool feature that I think was implemented for the Raspberry Pi but ultimately I'm not sure how useful it is.  I tried it out on my desktop just now.  It uses an SDL driver called KMSDRM basically letting Dosbox output graphics directly without an X server or Wayland compositor.  It works, even with OpenGL and CRT shader support.

```
SDL: version 2.26.5 initialised (KMSDRM video and pulseaudio audio)
```

It's a little janky with the resolution though.  On my system, it sets my 1080p monitor's resolution to 1152x864 which is a 4:3 resolution but my monitor (like many LCD panels) doesn't care about aspect ratio so it stretches to fill the 16:9 screen.  Maybe this mode is more useful to people that have CRT monitors.

I was also able to reproduce the "blackouts" you mentioned when the game switches to playing a video and back.  It's trying to change resolution on the host for some reason even though it ends up at the same 1152x864 on my screen.

This doesn't happen inside my KDE desktop environment.  I think you're likely to have a better experience running Dosbox inside an X server (or Wayland compositor).  Like I said, I do think this KMSDRM mode is cool but I'm not sure what the practical use-case is.
